docker/test/upgrade/run.sh

```diff
@@ -0,0 +1,200 @@
+# FIXME https://github.com/ClickHouse/ClickHouse/issues/39197 ("Missing columns: 'v3' while processing query: 'v3, k, v1, v2, p'")
+# FIXME https://github.com/ClickHouse/ClickHouse/issues/39174 - bad mutation does not indicate backward incompatibility
+echo "Check for Error messages in server log:"
+rg -Fav -e "Code: 236. DB::Exception: Cancelled merging parts" \
+           -e "Code: 236. DB::Exception: Cancelled mutating parts" \
+           -e "REPLICA_IS_ALREADY_ACTIVE" \
+           -e "REPLICA_ALREADY_EXISTS" \
+           -e "ALL_REPLICAS_LOST" \
+           -e "DDLWorker: Cannot parse DDL task query" \
+           -e "RaftInstance: failed to accept a rpc connection due to error 125" \
+           -e "UNKNOWN_DATABASE" \
+           -e "NETWORK_ERROR" \
+           -e "UNKNOWN_TABLE" \
+           -e "ZooKeeperClient" \
+           -e "KEEPER_EXCEPTION" \
+           -e "DirectoryMonitor" \
+           -e "TABLE_IS_READ_ONLY" \
+           -e "Code: 1000, e.code() = 111, Connection refused" \
+           -e "UNFINISHED" \
+           -e "NETLINK_ERROR" \
+           -e "Renaming unexpected part" \
+           -e "PART_IS_TEMPORARILY_LOCKED" \
+           -e "and a merge is impossible: we didn't find" \
+           -e "found in queue and some source parts for it was lost" \
+           -e "is lost forever." \
+           -e "Unknown index: idx." \
+           -e "Cannot parse string 'Hello' as UInt64" \
+           -e "} <Error> TCPHandler: Code:" \
+           -e "} <Error> executeQuery: Code:" \
+           -e "Missing columns: 'v3' while processing query: 'v3, k, v1, v2, p'" \
+           -e "The set of parts restored in place of" \
+           -e "(ReplicatedMergeTreeAttachThread): Initialization failed. Error" \
+           -e "Code: 269. DB::Exception: Destination table is myself" \
+           -e "Coordination::Exception: Connection loss" \
+           -e "MutateFromLogEntryTask" \
+           -e "No connection to ZooKeeper, cannot get shared table ID" \
+           -e "Session expired" \
+           -e "TOO_MANY_PARTS" \
+           -e "Authentication failed" \
+           -e "Container already exists" \
+    /var/log/clickhouse-server/clickhouse-server.upgrade.log | zgrep -Fa "<Error>" > /test_output/upgrade_error_messages.txt \
+    && echo -e "Error message in clickhouse-server.log (see upgrade_error_messages.txt)$FAIL$(head_escaped /test_output/bc_check_error_messages.txt)" \
+        >> /test_output/test_results.tsv \
+    || echo -e "No Error messages after server upgrade$OK" >> /test_output/test_results.tsv
+
+# Remove file upgrade_error_messages.txt if it's empty
+[ -s /test_output/upgrade_error_messages.txt ] || rm /test_output/upgrade_error_messages.txt
+
+# Grep logs for sanitizer asserts, crashes and other critical errors
+check_logs_for_critical_errors
+
+tar -chf /test_output/coordination.tar /var/lib/clickhouse/coordination ||:
+
+collect_query_and_trace_logs
+
+check_oom_in_dmesg
```

Comment:
@Avogar This has been never defined, and looks like it is obsolete, since everything is done in `check_logs_for_critical_errors`?

Reply:
Yes, just forget to remove it, thanks for noticing